ci/ray_ci/runner.py

```diff
@@ -77,24 +86,32 @@ def _chunk_into_n(list: List[str], n: int):
     return [list[x * size : x * size + size] for x in range(n)]
 
 
-def _get_all_test_query(targets: List[str], team: str) -> str:
+def _get_all_test_query(targets: List[str], team: str, size: str) -> str:
     test_query = " union ".join([f"tests({target})" for target in targets])
     team_query = f"attr(tags, team:{team}, {test_query})"
     size_query = " union ".join(
-        [f"attr(size, {s}, {test_query})" for s in ["small", "medium"]]
+        [f"attr(size, {s}, {test_query})" for s in size.split(",")]
+    )
+    except_query = " union ".join(
+        [
+            f"attr(tags, {t}, {test_query})"
+            for t in ["debug_tests", "asan_tests", "ray_ha"]
+        ]
     )
 
-    return f"{team_query} intersect ({size_query})"
+    return f"({team_query} intersect ({size_query})) except ({except_query})"
 
 
-def _get_all_test_targets(targets: str, team: str, yaml_dir: str) -> List[str]:
+def _get_all_test_targets(
+    targets: str, team: str, size: str, yaml_dir: str
+) -> List[str]:
```

Comment:
I feel that the filtering logic in these functions are getting a bit complicated now..

Reply:
It's an artifact of  tag usages in civ01 that makes unification rather complex. I hope to clean this up along the way.